docs/index.yml

```diff
@@ -0,0 +1,193 @@
+swagger: "2.0"
+info:
+  version: 1.0.0
+  title: YnvestTubeApi
+host: "127.0.0.1:8000"
+consumes:
+  - "application/json"
+produces:
+  - "application/json"
+paths:
+  /users:
+    get:
+      summary: 'Get all users'
+      tags:
+        - user
+      responses:
+        '200':
+          description: 'Users list retrieved successfully'
+          schema:
+            $ref: '#/definitions/UserList'
+        '403':
+          description: Bad user id
+  /users/register:
+    get:
+      summary: "Register new user"
+      tags:
+        - user
+      responses:
+        "200":
+          description: 'User id for an app to use in future calls'
+          schema:
+            $ref: '#/definitions/UserId'
+  /users/{userId}:
+    get:
+      summary: "Get user's overall info cash, account creation date."
+      tags:
+        - user
+      parameters:
+        - in: path
+          name: "UserId"
+          required: true
+          schema:
+            $ref: '#/definitions/UserId'
+      responses:
+        '200':
+          description: 'User data'
+          schema:
+            $ref: '#/definitions/UserDetails'
+        '403':
+          description: Bad user id
+  /users/{UserId}/details:
+    get:
+      summary: "Get user's actual auctions and all rentals/ transactions."
+      tags:
+        - user
+      parameters:
+        - in: path
+          name: "UserId"
+          required: true
+          schema:
+            $ref: '#/definitions/UserId'
+      responses:
+        '200':
+          description: 'User detailed data'
+          schema:
+            $ref: '#/definitions/UserDetails'
```

Comment:
Viewing the auctions attended by a user could be a nice feature, but it is not stated in requirements so if it would take any time to implement i would omit this field for now.